src/components/rooms/room-lobby.component.js

```diff
@@ -31,16 +35,30 @@ class RoomLobby extends React.Component {
         });
     }
 
+    showRoomCreate() {
+        this.setState({roomCreateVisible: true})
+    }
+
+    finishRoomCreate(newRoom) {
+        this.setState({roomCreateVisible: !this.state.roomCreateVisible}) 
```

Comment:
this.state.roomCreateVisible => use prevState.roomCreateVisible here? 
https://facebook.github.io/react/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous